src/core/server/plugins/discovery/plugin_manifest_parser.ts

```diff
@@ -175,7 +181,26 @@ export async function parseManifest(
     manifest.requiredOpenSearchPlugins.length > 0
   ) {
     log.info(
-      `Plugin ${manifest.id} has a dependency on following OpenSearch plugin(s): "${manifest.requiredOpenSearchPlugins}".`
+      `Plugin ${
+        manifest.id
+      } has a dependency on following OpenSearch plugin(s): [${manifest.requiredOpenSearchPlugins
+        .map((plugin) => plugin.id)
+        .join(', ')}].`
+    );
+  }
+
+  if (
+    Array.isArray(manifest.requiredOpenSearchPlugins) &&
+    manifest.requiredOpenSearchPlugins.length > 0 &&
+    !manifest.requiredOpenSearchPlugins.every((plugin) =>
+      isOpenSearchPluginVersionRangeValid(plugin.versionRange)
+    )
+  ) {
+    throw PluginDiscoveryError.invalidManifest(
+      manifestPath,
+      new Error(
+        `The "requiredOpenSearchPlugins" in plugin manifest for "${manifest.id}" has an invalid version range defined in one of the plugins.`
```

Comment:
In this current state I would suggest to make the version optional but with @AMoo-Miki suggestion on the other PR with "id":"version" will make sense and require both inputs.

Reply:
Uhum. As a user, what I would expect is
```js
{ ...
  "requiredOpenSearchPlugins": [
    "PluginA": "2.0.0",
    "PluginB": "^2.2.0",
    "PluginC": "^1.3.1 || ^2.5.1",
  ]
...}
```

Reply:
Sounds good! Updated